src/main/app/src/app/ontkoppelde-documenten/ontkoppelde-documenten-list/ontkoppelde-documenten-list.component.ts

```diff
@@ -0,0 +1,78 @@
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort} from '@angular/material/sort';
+import {merge} from 'rxjs';
+import {map, startWith, switchMap} from 'rxjs/operators';
+import {ListParameters} from '../../shared/model/list-parameters';
+import {InformatieObjectenService} from '../../informatie-objecten/informatie-objecten.service';
+import {MatTableDataSource} from '@angular/material/table';
+
+@Component({
+    templateUrl: './ontkoppelde-documenten-list.component.html',
+    styleUrls: ['./ontkoppelde-documenten-list.component.less']
+})
+export class OntkoppeldeDocumentenListComponent implements OnInit, AfterViewInit {
+
+    isLoadingResults = true;
+    dataSource: MatTableDataSource<OntkoppeldDocument> = new MatTableDataSource<OntkoppeldDocument>();
+    @ViewChild(MatPaginator) paginator: MatPaginator;
+    @ViewChild(MatSort) sort: MatSort;
+    displayedColumns: string[] = ['documentID', 'creatiedatum', 'titel', 'actions'];
+    defaults: ListParameters;
+
+    constructor(private service: OntkoppeldeDocumentenService,
+                private infoService: InformatieObjectenService,
+                private utilService: UtilService,
+                private sessionStorageService: SessionStorageService) { }
+
+    ngOnInit(): void {
+        this.utilService.setTitle('title.documenten.ontkoppeldeDocumenten');
+        this.defaults = this.sessionStorageService.getSessionStorage('ontkoppeldeDocumenten', new ListParameters('documentID', 'desc')) as ListParameters;
+    }
+
+    ngAfterViewInit(): void {
+        this.sort.sortChange.subscribe(() => (this.paginator.pageIndex = 0));
+        merge(this.sort.sortChange, this.paginator.page).pipe(
+            startWith({}),
+            switchMap(() => {
+                this.isLoadingResults = true;
+                this.utilService.setLoading(true);
+                return this.service.list(this.getListParameters());
+            }),
+            map(data => {
+                this.isLoadingResults = false;
+                this.utilService.setLoading(false);
+                this.sessionStorageService.setSessionStorage('ontkoppeldeDocumenten', this.getListParameters());
+                return data;
+            })
+        ).subscribe(data => {
+            this.paginator.length = data.count;
+            this.dataSource.data = data.results;
+        });
+    }
+
+    getListParameters(): ListParameters {
+        const params = new ListParameters(this.sort.active, this.sort.direction);
+        params.page = this.paginator.pageIndex;
+        params.maxResults = this.paginator.pageSize;
+        return params;
+    }
+
+    getDownloadURL(od: OntkoppeldDocument): string {
+        return this.infoService.getDownloadURL(od.documentUUID);
+    }
+
+    reset(): void {
+        this.sessionStorageService.setSessionStorage('ontkoppeldeDocumenten', new ListParameters('documentID', 'desc'));
+        this.utilService.reloadRoute();
```

Comment:
Is dit wel nodig, kun je niet gewoon een init opnieuw aanroepen?

Reply:
dan zou ik het ophalen van de data anders moeten doen, maar het inladen van de view is niet een hele zware opertatie en weet je zeker dat alles wordt gereset